core/modules/replica/HttpProcessor.cc

```diff
@@ -1226,6 +1303,61 @@ void HttpProcessor::_addTableConfig(qhttp::Request::Ptr req,
 }
 
 
+void HttpProcessor::_sqlQuery(qhttp::Request::Ptr req,
+                              qhttp::Response::Ptr resp) {
+    _debug(__func__);
+
+    try {
+
+        // All parameters must be provided via the body of the request
+
+        ::HttpRequestBody body(req);
+
+        auto const worker   =        body.required("worker");
+        auto const query    =        body.required("query");
+        auto const user     =        body.required("user");
+        auto const password =        body.required("password");
+        auto const maxRows  = stoull(body.optional("max_rows", "0"));
+
+        _debug(string(__func__) + " worker="   +           worker);
+        _debug(string(__func__) + " query="    +           query);
+        _debug(string(__func__) + " user="     +           user);
+        _debug(string(__func__) + " password=" +           password);
+        _debug(string(__func__) + " maxRows="  + to_string(maxRows));
+
+        atomic<bool> finished{false};
+        auto const request = controller()->sql(
+            worker,
+            query,
+            user,
+            password,
+            maxRows,
+            [&finished] (SqlRequest::Ptr const& request) {
+                finished = true;
+            }
+        );
+        util::BlockPost blockPost(10, 20);  // for random delays (in milliseconds) between iterations
+        while (not finished) {
+            blockPost.wait();
+        }
```

Comment:
Making the async requests and jobs classes synchronous would be messy. Though, In theory I could introduce wrapper methods which would launch operations on the original async `*Job` or `*Request` classes  with a custom callback, and then wait for completion of the job/request using `cv` or that BlockPost (in case if I need heartbeats). I will put this side onto my _TO DO_ list.

Reply:
I've put the code improvement proposal still under development) here:
https://confluence.lsstcorp.org/pages/viewpage.action?pageId=102471961